### PR TITLE
Enhancements and polish

### PR DESCRIPTION
**Pull request by tsalb**
`soqlDatatable` updates:
- Add `iconName`. See help tooltip.
- Improve api initiated `refresh` UX (spinners now work better).
- Add `composedTitle` slot for putting custom components inside a `lightning-card` base component `title` slot.
- Remove case sensitivity for `checkboxType`.
- Clean `selectedRows` output when passing to a Flow table action.
    - Seems Summer 20 made some changes to how Flow ingests a serialized `sObject` inputVariable.
    - This doesn't fix it, but it does strip the flattened (parent) tableData prior to sending, just in case.

`messageService` and related component updates:
- Updated with an `open` variant of some APIs which will bypass any `boundaries` set up.
    - This supports better dialog close behavior with "Tab Focused Dialogs" release update.
    - See Event Pulbisher [examples](https://github.com/tsalb/lwc-utils/wiki/EventPublisher#examples) for more detail.

### CODE REVIEW

**Comment by tsalb** [comment]
FYI @jongpie - made some changes to your commits [here](https://github.com/tsalb/lwc-utils/pull/100/commits/b99ad0e521464619099b24d605e0e615519d6d97).

**tl;dr;** Moved icon logic to `baseDatatable` for future updates to `collectionDatatable` (one day).
